pkg/nuctl/command/patch.go

```diff
@@ -0,0 +1,366 @@
+
+	// initialize http client
+	c.httpClient = &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+
+	if err := c.initializePatchOptions(); err != nil {
+		return errors.Wrap(err, "Failed to initialize patch options")
+	}
+
+	c.rootCommandeer.loggerInstance.DebugWith("Initialized patch commandeer",
+		"patchOptions", c.patchOptions)
+
+	return nil
+}
+
+func (c *patchCommandeer) initializePatchOptions() error {
+	if len(c.patchOptionsMap) == 0 {
+		return nil
+	}
+
+	c.patchOptions = &resource.PatchOptions{}
+
+	// convert the patch options map to a patch options struct
+	if err := mapstructure.Decode(c.patchOptionsMap, c.patchOptions); err != nil {
+		return errors.Wrap(err, "Failed to decode patch options")
+	}
+
+	return nil
+}
+
+type patchFunctionsCommandeer struct {
+	*patchCommandeer
+
+	excludedProjects   []string
+	excludedFunctions  []string
+	concurrency        int
+	username           string
+	password           string
+	apiURL             string
+	waitForFunction    bool
+	outputManifest     map[string]interface{}
+	outputManifestLock sync.Mutex
+}
+
+func newPatchFunctionsCommandeer(ctx context.Context, patchCommandeer *patchCommandeer) *patchFunctionsCommandeer {
+	commandeer := &patchFunctionsCommandeer{
+		patchCommandeer: patchCommandeer,
+		outputManifest: map[string]interface{}{
+			"Failed":  map[string]error{},
+			"Success": []string{},
+		},
+		outputManifestLock: sync.Mutex{},
+	}
+
+	cmd := &cobra.Command{
+		Use:     "functions",
+		Aliases: []string{"func", "fn", "function"},
+		Short:   "patch functions",
+		RunE: func(cmd *cobra.Command, args []string) error {
+
+			// initialize root
+			if err := patchCommandeer.rootCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize root")
+			}
+
+			// initialize patch commandeer
+			if err := patchCommandeer.initialize(); err != nil {
+				return errors.Wrap(err, "Failed to initialize patch commandeer")
+			}
+
+			return commandeer.patchFunctions(ctx)
+		},
+	}
+
+	cmd.PersistentFlags().StringSliceVarP(&commandeer.excludedProjects, "exclude-projects", "", []string{}, "Exclude projects to patch")
+	cmd.PersistentFlags().StringSliceVarP(&commandeer.excludedFunctions, "exclude-functions", "", []string{}, "Exclude functions to patch")
+	cmd.PersistentFlags().IntVarP(&commandeer.concurrency, "concurrency", "c", 10, "Max number of parallel patches")
+	cmd.PersistentFlags().StringVarP(&commandeer.username, "username", "u", "", "Username of a user with permissions to patch functions")
+	cmd.PersistentFlags().StringVarP(&commandeer.password, "password", "p", "", "Password of a user with permissions to patch functions")
+	cmd.PersistentFlags().StringVarP(&commandeer.apiURL, "api-url", "", "", "URL of the nuclio API (e.g. https://nuclio.io:8070)")
+	cmd.PersistentFlags().BoolVarP(&commandeer.waitForFunction, "wait", "w", false, "Wait for function to be ready after patching")
+
```

Comment:
add flag `skip-functions-with-gpu` to allow skipping function requiring gpus as those are expensive.
in such case, just log them out to the manifest of "skipped" functions